### PR TITLE
BAVL-412 and postgres test container so can run tests without having to run docker compose file.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
If a container is already running then it will use it, if not a test container will be started.

This was bugging me a little, if I ran the tests in Intellij I had to make sure was running Postgres container manually.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/config/FeatureSwitchesTest.kt [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.FeatureSwitches
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isBool
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.integration.IntegrationTestBase
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.InboundEventsListener
 import uk.gov.justice.hmpps.sqs.HmppsQueueService
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
-@ActiveProfiles("test")
-class FeatureSwitchesTest {
+class FeatureSwitchesTest : IntegrationTestBase() {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/config/FeatureSwitchesTest.kt:17]: Moved as it needed a running application anyway.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt [hunk 2]
```diff
@@ -122,4 +125,18 @@ abstract class IntegrationTestBase {
...
+
+    @JvmStatic
+    @DynamicPropertySource
+    fun properties(registry: DynamicPropertyRegistry) {
+      pgContainer?.run {
+        registry.add("spring.datasource.url", pgContainer::getJdbcUrl)
+        registry.add("spring.datasource.username", pgContainer::getUsername)
+        registry.add("spring.datasource.password", pgContainer::getPassword)
+      }
+    }
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt:140]: Will this start pg container for all tests, whether needed or not? 
In contacts, we had two base classes .. one for PostgresInt Test and another of H2

**Review by timharrison-moj — APPROVED** [review]
LGTM. Small question

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt:140]: There is no H2 anymore.  Steve took it out, it is using a real DB.

**Pull request merged**